nifi-nar-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/src/main/resources/docs/org.apache.nifi.processors.salesforce.QuerySalesforceObject/additionalDetails.html

```diff
@@ -0,0 +1,73 @@
+      The ASF licenses this file to You under the Apache License, Version 2.0
+      (the "License"); you may not use this file except in compliance with
+      the License.  You may obtain a copy of the License at
+          http://www.apache.org/licenses/LICENSE-2.0
+      Unless required by applicable law or agreed to in writing, software
+      distributed under the License is distributed on an "AS IS" BASIS,
+      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+      See the License for the specific language governing permissions and
+      limitations under the License.
+    -->
+
+<head>
+    <meta charset="utf-8"/>
+    <title>QuerySalesforceObject</title>
+    <link rel="stylesheet" href="../../../../../css/component-usage.css" type="text/css"/>
+    <style>
+h2 {margin-top: 4em}
+h3 {margin-top: 3em}
+td {text-align: left}
+    </style>
+</head>
+
+<body>
+
+<h1>QuerySalesforceObject</h1>
+
+<h3>Description</h3>
+
+<p>
+    Objects in Salesforce are database tables, their rows are known as records, and their columns are called fields. The QuerySalesforceObject processor queries Salesforce objects and retrieves their records.
+    The processor constructs the query using SOQL (Salesforce Object Query Language) and retrieves the result record dataset using the Salesforce REST API.
+    The processor utilizes streams and NiFi record-based processing to be able to handle a large number of records and to allow arbitrary output format.
+</p>
+
+<h3>OAuth2 Access Token Provider Service</h3>
+
+<p>
+    The Salesforce REST API authorization is handled by the OAuth2 Access Token Provider Service. In order to use OAuth2 authorization create a new StandardOauth2AccessTokenProvider service and configure it
+    the following way.
+</p>
+
+<ul>
+    <li>Authorization Server URL: It is the concatenation of the base URL and the token request service URL (/services/oauth2/token).</li>
+    <li>Grant Type: User Password.</li>
+    <li>Username: The email address registered in the Salesforce account in use.</li>
+    <li>Password: For the Password a Security token must be requested. Go to Profile -> Settings and under the Reset My Security Token option, request one, which will be emailed to you.
+        The password is made up of the Salesforce account password and the Security token concatenated together without a space.</li>
+    <li>Client ID: Create a new Connected App within Salesforce. Go to Setup -> On the left search panel find App Manager -> Create New Connected App. Once it’s done, the Consumer Key goes to the Client ID property.</li>
+    <li>Client Secret: It can be found on the same Connected App page under the name of Consumer Secret.</li>
+</ul>
+
+<h3>Age properties</h3>
+
+<p>
+    The age property is important if it is intended to query and process a record once. It provides a sliding window that starts with the processor’s prior run time and ends with the current run time minus the age delay.
```

Comment:
Recommend the following adjustments:
```suggestion
    The age properties are important to avoid processing duplicate records. Age filtering provides a sliding window that starts with the processor’s prior run time and ends with the current run time minus the age delay.
```